rust/flatbuffers/src/builder.rs

```diff
@@ -0,0 +1,604 @@
+        //
+        // Layout of the data this function will create when we re-use an
+        // existing vtable.
+        //
+        // We always serialize this particular vtable, then compare it to the
+        // other vtables we know about to see if there is a duplicate. If there
+        // is, then we erase the serialized vtable we just made.
+        // We serialize it first so that we are able to do byte-by-byte
+        // comparisons with already-serialized vtables. This 1) saves
+        // bookkeeping space (we only keep revlocs to existing vtables), 2)
+        // allows us to convert to little-endian once, then do
+        // fast memcmp comparisons, and 3) by ensuring we are comparing real
+        // serialized vtables, we can be more assured that we are doing the
+        // comparisons correctly.
+        //
+        // --------------------------------------------------------------------
+        // table starts here
+        // | x, x, x, x -- offset (negative direction) to an existing vtable [i32]
+        // |               aka "vtableoffset"
+        // | -- table inline data begins here, we don't touch it --
+        // table starts here: aka "table_start"
+        // --------------------------------------------------------------------
+
+        // Include space for the last offset and ensure empty tables have a
+        // minimum size.
+        let max_voffset = self.field_locs.iter().map(|fl| fl.id).max();
+        let vtable_len = match max_voffset {
+            None => { field_index_to_field_offset(0) as usize }
+            Some(mv) => { mv as usize + SIZE_VOFFSET }
+        };
+        self.fill(vtable_len);
+        let table_object_size = object_vtable_revloc.value() - table_tail_revloc.value();
+        debug_assert!(table_object_size < 0x10000); // Vtable use 16bit offsets.
+
+        let vt_start_pos = self.head;
+        let vt_end_pos = self.head + vtable_len;
+        {
+            let vtfw = &mut VTableWriter::init(&mut self.owned_buf[vt_start_pos..vt_end_pos]);
+            vtfw.write_vtable_byte_length(vtable_len as VOffsetT);
+            vtfw.write_object_inline_size(table_object_size as VOffsetT);
+            for &fl in self.field_locs.iter() {
+                let pos: VOffsetT = (object_vtable_revloc.value() - fl.off) as VOffsetT;
+                debug_assert_eq!(vtfw.get_field_offset(fl.id),
+                                 0,
+                                 "tried to write a vtable field multiple times");
+                vtfw.write_field_offset(fl.id, pos);
+            }
+        }
+        let vt_use = {
+            let mut ret: usize = self.used_space();
+
+            // LIFO order
+            for &vt_rev_pos in self.written_vtable_revpos.iter().rev() {
+                let eq = {
+                    let this_vt = VTable::init(&self.owned_buf[..], self.head);
+                    let other_vt = VTable::init(&self.owned_buf[..], self.head + self.used_space() - vt_rev_pos as usize);
```

Comment:
do Rust people have a typical line limit?

Reply:
I think `rustfmt` generally wraps at around 80 characters

Reply:
This isn't generated code.
For actually generated code, we don't support line-limits.

Reply:
Sorry, I could have said "generate or write". A lot of folks seem to rely on editor tools when writing it to automatically format on save. That allows projects to place their own rustfmt configurations in repos and expect contributors to have run rustfmt when submitting PRs.